core/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/BalancedShardsAllocator.java

```diff
@@ -187,19 +192,19 @@ public WeightFunction(float indexBalance, float shardBalance) {
         }
 
         public float weight(Balancer balancer, ModelNode node, String index) {
-            return weight(balancer, node, index, 0);
+            return weight(balancer, node, index, 0, 0.0f);
```

Comment:
the actual value does not matter, it is multiplied by the previous parameter (which is 0). To make it less confusing, I'm using the default (1.0) now.